jinahub/indexers/dbms/FileDBMSIndexer/__init__.py

```diff
@@ -334,3 +334,23 @@ def physical_size(self) -> int:
         """
         root_directory = Path(self.workspace)
         return sum(f.stat().st_size for f in root_directory.glob('**/*') if f.is_file())
+
+    def _get(self, docs: DocumentArray, **kwargs):
+        """Get a document by its id
+
+        :param docs: the documents
+        """
+        self._search(docs, is_update=False)
+
+    def _search(self, docs: DocumentArray, is_update):
+        for i, doc in enumerate(docs):
+            vec, doc_bytes = pickle.loads(self._query([doc.id])[0])
+            serialized_doc = Document(doc_bytes)
+            serialized_doc.pop('content_hash')
+            if is_update:
```

Comment:
Just asking, is it still needed to differentiate both? it leads to a very ugly pattern. (I guess is the problem with `shapes` and stuff right?)

Reply:
Yeah, I guess. Not to be addressed in this PR. I only touched this file to add a private reading method (which is just C&P form the FileQueryIndexer)